src/core/public/application/integration_tests/application_service.test.tsx

```diff
@@ -56,6 +57,69 @@ describe('ApplicationService', () => {
     service = new ApplicationService();
   });
 
+  describe('navigating to apps', () => {
+    describe('using history.push', () => {
+      it('emits currentAppId$ before mounting the app', async () => {
```

Comment:
Added tests to ensure consistency between `history.push` and `application.navigateToApp`

Reply:
> history.push is exposed to plugins indirectly via ScopedHistory

Yea, but scoped history is scoped to the mounted app's path, right? So how would you test pushing to another app's path to ensure the appId changed?